infra/gcp/ensure-main-project.sh

```diff
@@ -84,6 +89,45 @@ enable_api "${PROJECT}" secretmanager.googleapis.com
 color 6 "Ensuring the cluster terraform-state bucket exists"
 ensure_private_gcs_bucket "${PROJECT}" "gs://${CLUSTER_TERRAFORM_BUCKET}"
 
+
+color 6 "Ensuring all the prow buckets exist"
+for bucket in "${PROW_BUCKETS[@]}"; do
+    color 6 "Ensuring bucket ${bucket} exists."
+    ensure_public_gcs_bucket "${PROJECT}" "gs://${bucket}"
+
+    SERVICE_ACCOUNT_NAME="sa-${bucket}"
+    SERVICE_ACCOUNT_EMAIL="$(svc_acct_email "${PROJECT}" \
+        "${SERVICE_ACCOUNT_NAME}")"
+    SECRET_ID="${SERVICE_ACCOUNT_NAME}-key"
+    TMP_DIR=$(mktemp -d "/tmp/${SERVICE_ACCOUNT_NAME}.XXXXXX")
+    KEY_FILE="${TMP_DIR}/key.json"
+
+    color 6 "Creating service account: ${SERVICE_ACCOUNT_NAME}"
+    ensure_service_account \
+        "${PROJECT}" \
+        "${SERVICE_ACCOUNT_NAME}" \
+        "${SERVICE_ACCOUNT_NAME}"
+
+    color 6 "Empowering service account: ${SERVICE_ACCOUNT_NAME}"
+    empower_svcacct_to_write_gcs_bucket "${SERVICE_ACCOUNT_EMAIL}" "gs://${bucket}"
+
+    color 6 "Creating private key for service account: ${SERVICE_ACCOUNT_NAME}"
+    gcloud iam service-accounts keys create "${KEY_FILE}" \
+        --project "${PROJECT}" \
+        --iam-account "${SERVICE_ACCOUNT_EMAIL}"
+
+    color 6 "Creating secret to store private key"
```

Comment:
Who can access the secret? I would expect to see a `gcloud secrets add-iam-policy-binding` call

My suggestion would be k8s-infra-prow-oncall@ (I'm happy to approve a PR adding you as a member to that group)

Reply:
I'll push the binding once the right option is chosen for `gs://kubernetes-jenkins` migration.